llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -588,9 +588,10 @@ def tensor_hook(
                     del new_base_model_instance
                 else:
                     new_model_instance = type(original_model)(new_config)
-                    new_model_instance.generation_config.update(
-                        **original_model.generation_config.to_dict(),
-                    )
+                    if new_model_instance.generation_config:
```

Comment:
this probably isn't the right check. I think if it is empty dict you still want to update it. also please add a test for this.